### ISSUE TITLE
Filter Configuration - Development

### ISSUE THREAD

**Issue by jamlung-ri** [OpenConceptLab/ocl_issues#574]
Once view configuration is complete in the Metadata Browser, then user-facing filters can be configured within those views to do the following:
* Create custom filters (with selectable options only, no text entry or other functionality for now)
* Hide OCL’s default filters
* Define an order/arrangement for all filters

Details for this can be found here: https://docs.google.com/presentation/d/1xdZvvgWAB8yem4xLWC5kMOouQHeFE_c3F4cR2uQBDXo/edit#slide=id.gb9293fe012_0_0

**Comment by snyaggarwal** [comment]
@jamlung-ri -- 
I tried to implement user-filters with options to interact with. 

Two types of filters supported now:
1. _(existing)_ Context filters -- Always applied, cannot remove, cannot interact
  - Example: `filters: {concept_class: 'Diagnosis,Finding'}`
2. **(new)** User Filters -- Interactive filters. Dropdown with list of options to choose from.
  - Example: 
  ```
    filters: {
        'extras.classification': {
          name: 'Classification',  (optional, if not provided "Extras Classification" will become name)
          tooltip: 'Filter by Classification' (optional, if not provided a computed tooltip based on name will be applied)
          options: [
            'fine',
            'coarse',
            'n/a'
          ]
        }
      }
   ```

###### Below is the example configuration for PEPFAR source:
```
{
  tabs: [
    {
      type: 'concepts',
      label: 'Concepts',
      layout: 'table',
      default: true,
      filters: {
        'extras.classification': {
          name: 'Classification',
          options: [
            'fine',
            'coarse',
            'n/a'
          ]
        }
      },
      page_size: 25
    },
    {
      type: 'mappings',
      label: 'Mappings',
      layout: 'table',
      page_size: 25
    },
    {
      type: 'concepts',
      label: 'Problems',
      layout: 'List',
      default: false,
      filters: {
        concept_class: 'Diagnosis,Finding'
      },
      page_size: 25
    },
    {
      type: 'concepts',
      label: 'Drugs',
      layout: 'List',
      default: false,
      filters: {
        concept_class: 'Drug'
      },
      page_size: 25
    },
    {
      type: 'versions',
      label: 'Versions',
      layout: 'table',
      page_size: 25
    },
    {
      type: 'about',
      label: 'About'
    }
  ]
}
```

**Comment by jamlung-ri** [comment]
This is looking really great! I was able to create my own filters, in addition to the existing context filters, in a PEPFAR config file. Example here: https://app.qa.aws.openconceptlab.org/#/orgs/PEPFAR-Test10b/?configs=true ; in the Code Lists tab, you can see the Type and Period filters.

Notes for the future (might add to this later):
* We should be able to support multi-select filters if specified by the user.

**Comment by jamlung-ri** [comment]
@snyaggarwal Would it be easy to enable a "default" for a certain filter option? For example, for the Period filter in the example above, it would be useful for FY21 to be the default value, but still allow the user to change that filter to Fy20, FY16, etc.

This isn't super high priority but worth suggesting.

**Comment by jamlung-ri** [comment]
@snyaggarwal One issue is that only a single filter can be applied at one time. For example, in the Reference Indicator tab of the MSP config here: https://app.qa.aws.openconceptlab.org/#/orgs/PEPFAR-Test10b/sources/MER/?configs=true
I could only apply the Type or the Period filter, but not both.

This is something that does not need to be done immediately but probably in the next few weeks or so.

**Comment by snyaggarwal** [comment]
> @snyaggarwal Would it be easy to enable a "default" for a certain filter option? For example, for the Period filter in the example above, it would be useful for FY21 to be the default value, but still allow the user to change that filter to Fy20, FY16, etc.
> 
> This isn't super high priority but worth suggesting.

@jamlung-ri You can define "default" value for a filter by following:
```
filters: {
        'extras.classification': {
          name: 'Classification',
          default: 'coarse',
          options: [
            'fine',
            'coarse',
            'n/a'
          ]
        }
      }
      ```

**Comment by snyaggarwal** [comment]
> @snyaggarwal One issue is that only a single filter can be applied at one time. For example, in the Reference Indicator tab of the MSP config here: https://app.qa.aws.openconceptlab.org/#/orgs/PEPFAR-Test10b/sources/MER/?configs=true
> I could only apply the Type or the Period filter, but not both.
> 
> This is something that does not need to be done immediately but probably in the next few weeks or so.

@jamlung-ri this should be fixed now

**Comment by jamlung-ri** [comment]
I recorded the future suggestions in another ticket, so this one can be closed unless something immediate comes up.

@snyaggarwal This ticket does not have an estimate, in case you want to add one.
